File: [프로그래머스] 코딩테스트 입문_합성수 찾기.js
function solution(n) {
    var answer = 0;
    for(let i = 4; i <= n; i++){
        let cnt = 1
        for(let j = 2; j <= n; j++){
            if(i % j === 0) { 
                cnt++;
            }
            if(cnt >= 3){
                answer++;
                break;
            } 
        }
    }
    return n <= 3 ? 0 :answer;
}

----------------------------------

function solution(n) {
    // i와 j 사이에 약수가 하나라도 존재하면 합성수
    var answer = 0;
    for(let i = 4; i <= n; i++){
        for(let j = 2; j < i; j++){
            if(i % j === 0) { 
                answer++;
                break;
            } 
        }
    }
    return answer;
}

File: [프로그래머스] 코딩테스트 입문_합성수 찾기.py
def solution(n):
    answer = 0
    for i in range(4, n+1):
        j = 1
        cnt = 1
        while i >= j:
            if cnt == 3: 
                answer += 1
                break
            if i % j == 0:
                cnt += 1
            j += 1
    return answer